Adds changelog for 12.1.3 (#7587)
citusdata/citus: CHANGELOG.md

### citus v12.1.3 (April 18, 2024) ###

* Allows overwriting host name for all inter-node connections by supporting "host"
  parameter in citus.node_conninfo (#7541)

> * Correctly mark some variables as volatile (#7570)

Reviewer: We use present simple tense to start description entries:

```suggestion
* Fixes some potential bugs by correctly marking some variables as volatile (#7570)
```

Author: Added as you suggested

> * Fix performance issue in GetForeignKeyOids on systems with many constraints (#7578)

Reviewer: Better to have a more user friendly description --users are not expected to know underlying C function names:

```suggestion
* Fixes performance issue in when tracking foreign key constraints on systems with many constraints (#7578)
```

Author: Added as you suggested

> * Fix performance issue when creating distributed tables and many already exist (#7575)
> 
> * Fix performance issue when creating distributed tables if many already exist (#7579)

Reviewer: let's combine them to one

```suggestion
* Fixes a performance issue when creating distributed tables if many already exist (#7575, #7579)
```

Author: Added as you suggested

* Fix performance issue when distributing a table that depends on an extension (#7574)

* Fix performance issue when using "\d tablename" on a server with many tables (#7577)

* Fixes a crash caused by some form of ALTER TABLE ADD COLUMN statements.
  When adding multiple columns, if one of the ADD COLUMN statements contains a FOREIGN

>   constraint ommitting the referenced columns in the statement, a SEGFAULT occurs. (#7522)

Reviewer: ```suggestion
  constraint omitting the referenced columns in the statement, a SEGFAULT was occurring. (#7522)
```

Author: Added as you suggested

> * Fixes invalid copy-paste rightComparisonLimit (#7547)

Reviewer: let's delete this because in-practice this doesn't cause user-facing issue: https://github.com/citusdata/citus/issues/7543#issuecomment-1975242612

Author: Deleted

> * Log username in the failed connection message (#7432)

Reviewer: ```suggestion
* Logs username in the failed connection message (#7432)
```

Author: Added as you suggested

### citus v12.1.3 (April 18, 2024) ###

* Allows overwriting host name for all inter-node connections by
  supporting "host" parameter in citus.node_conninfo (#7541)

* Changes the order in which the locks are acquired for the target and
  reference tables, when a modify request is initiated from a worker
  node that is not the "FirstWorkerNode" (#7542)

> * Fixes performance issue when distributing a table that depends on an

Reviewer 2: ```suggestion
* Fixes a performance issue when distributing a table that depends on an
```

  extension (#7574)

> * Fixes performance issue when using "\d tablename" on a server with

Reviewer 2: ```suggestion
* Fixes a performance issue when using "\d tablename" on a server with
```

  many tables (#7577)

* Fixes a crash caused by some form of ALTER TABLE ADD COLUMN

>   statements.When adding multiple columns, if one of the ADD COLUMN

Reviewer: ```suggestion
  statements. When adding multiple columns, if one of the ADD COLUMN
```

  statements contains a FOREIGN constraint omitting the referenced
  columns in the statement, a SEGFAULT was occurring. (#7522)

* Fixes a performance issue when creating distributed tables if many
  already exist (#7575, #7579)

> * Fixes bug when hostname in pg_dist_node resolves to multiple IPs

Reviewer: ```suggestion
* Fixes a bug when hostname in pg_dist_node resolves to multiple IPs
```

  (#7377)

> * Fixes performance issue in when tracking foreign key constraints on

Reviewer 2: ```suggestion
* Fixes a performance issue in when tracking foreign key constraints on
```

> * Fixes performance issue in when tracking foreign key constraints on
>   systems with many constraints (#7578)

Reviewer: ```suggestion
* Fixes performance issue when tracking foreign key constraints on
  systems with many constraints (#7578)
```

> * Fixes segmentation fault when using CASE WHEN in DO block functions
>   (#7554)

Reviewer: ```suggestion
* Fixes segmentation fault when using CASE WHEN in DO block within
  functions. (#7554)
```
